tests/integration/test.changes.js

```diff
@@ -343,24 +360,43 @@ adapters.forEach(function (adapter) {
           views: {
             even: {
               map: 'function (doc) { if (doc.integer % 2 === 1)' +
-               ' { emit(doc._id, null) }; }'
+                ' { emit(doc._id, null) }; }'
             }
           }
         }
       ];
-      var db = new PouchDB(dbs.name);
-      testUtils.writeDocs(db, docs, function () {
-        db.changes({
-          filter: 'foo/even',
-          limit: 2,
-          include_docs: true
-        }).on('error', function (err) {
-          err.status.should.equal(testUtils.errors.MISSING_DOC.status,
-                                  'correct error status returned');
-          err.name.should.equal('not_found');
-          done();
-        });
-      });
+      const caughtErrors = [];
+      const db = new PouchDB(dbs.name);
+
+      testUtils.promisify(testUtils.writeDocs)(db, docs)
+        .then(async () => {
+          try {
+            await db.changes({
```

Comment:
What does the `await` do here?  Is it equivalent to using the `complete` event?

Reply:
Indeed, if i run the test without the `await` keyword, the test didn't pass inside the function and therefore a warning (uncaught exception) was thrown.
As a bonus: the `await` is also a guard to ensure that other exceptions (if any) get caught, so we know when an unhandled side effect pollutes this test.